tests/acceptance/00_basics/03_bodies/inherit_from.cf

```diff
@@ -0,0 +1,54 @@
+##############################################################################
+#
+# Redmine #4309: body inheritance with inherit_from
+#
+##############################################################################
+
+body common control
+{
+      inputs => { "../../default.cf.sub" };
+      bundlesequence => { default("$(this.promise_filename)") };
+      version => "1.0";
+}
+
+bundle agent init
+{
+}
+
+bundle common test
+{
+  reports:
+      "test_inherit" classes => classes_generic_inherit("test_inherit");
+      "test_nonesuch" classes => classes_generic_nonesuch("test_nonesuch");
+      "test_overwrite" classes => classes_generic_overwrite("test_overwrite");
+}
+
+body classes classes_generic_inherit(x)
+{
+      inherit_from => classes_generic;
```

Comment:
OK, it's not done, but I pushed a commit with tests and 2 TODO items in `promises.c` that collects the given argument count and when it's done, will check it against the body prototype.  Check that the tests match what you expect, please.